content/webapp/components/VenueHours/VenueHours.test.tsx

```diff
@@ -0,0 +1,210 @@
+  overrideType: 'Christmas and New Year',
+  opens: '10:00',
+  closes: '18:00',
+  isClosed: false,
+};
+
+const closedExceptionalMonday: ExceptionalOpeningHoursDay = {
+  overrideDate: new Date('2022-12-26T00:00:00.000Z'),
+  overrideType: 'Christmas and New Year',
+  opens: '00:00',
+  closes: '00:00',
+  isClosed: true,
+};
+
+describe('UnusualOpeningHours', () => {
+  describe('Case 1: the venue is closed, and would have been on a regular day', () => {
+    it('renders a regular “Closed” time if the venue was already closed this day', () => {
+      const component = shallow(
+        <UnusualOpeningHours
+          regular={closedMonday}
+          exceptional={closedExceptionalMonday}
+        />
+      );
+
+      expect(
+        component.matchesElement(
+          <>
+            <OverrideDay>Monday 26 December</OverrideDay>
+            Closed
+          </>
+        )
+      ).toBe(true);
+    });
+  });
+
+  describe('Case 2: the venue is closed, and wouldn’t be on a regular day', () => {
+    it('marks “Closed” as unusual if the venue isn’t usually closed this day', () => {
+      const component = shallow(
+        <UnusualOpeningHours
+          regular={openMonday}
+          exceptional={closedExceptionalMonday}
+        />
+      );
+
+      expect(
+        component.matchesElement(
+          <>
+            <OverrideDay>Monday 26 December</OverrideDay>{' '}
+            <DifferentToRegularTime>Closed</DifferentToRegularTime>
+          </>
+        )
+      ).toBe(true);
+    });
+  });
+
+  describe('Case 3: the venue is open, but would normally be closed', () => {
+    it('marks the opening times as unusual if the venue isn’t usually open', () => {
+      const component = shallow(
+        <UnusualOpeningHours
+          regular={closedMonday}
+          exceptional={openExceptionalMonday}
+        />
+      );
+
+      expect(
+        component.matchesElement(
+          <>
+            <OverrideDay>Monday 26 December</OverrideDay>{' '}
+            <DifferentToRegularTime>10:00 – 18:00</DifferentToRegularTime>
+          </>
+        )
+      ).toBe(true);
+    });
+  });
+
+  describe('Case 4: the venue is open', () => {
+    it('doesn’t mark the opening times as unusual if they match a regular day', () => {
+      const component = shallow(
+        <UnusualOpeningHours
+          regular={openMonday}
+          exceptional={openExceptionalMonday}
+        />
+      );
+
+      // This and subsequent tests are a bit fragile and weird; for some reason the
+      // matchesElement() approach I used in the other tests doesn't work here.
+      expect(component.debug()).toBe(`<Fragment>
+  <VenueHours__OverrideDay>
+    Monday
+     
+    26 December
+  </VenueHours__OverrideDay>
+   
+  10:00
+   – 
+  18:00
+</Fragment>`);
+
+      //   expect(
+      //     component.matchesElement(
+      //       <>
+      //         <OverrideDay>Monday 26 December</OverrideDay> {'10:00 – 18:00'}
```

Comment:
Are the curly braces what was preventing `matchesElement` from working, maybe?

Reply:
I'm not sure, I don't think so – I added those in an attempt to get it working, but it didn't seem to make a difference.

Reply:
fair enough